sonar-helm-for-iac/main.go

```diff
@@ -1,9 +1,92 @@
+// SonarQube IaC Plugin
+// Copyright (C) 2018-2023 SonarSource SA
+// mailto:info AT sonarsource DOT com
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 3 of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program; if not, write to the Free Software Foundation,
+// Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
 package main
 
+import "C"
 import (
 	"fmt"
+	"strings"
+	"text/template"
+
+	"sigs.k8s.io/yaml"
 )
 
 func main() {
 	fmt.Println("Hello World!")
 }
+
+var handles []*template.Template
+
+//export EvaluateTemplate
+func EvaluateTemplate(path string, content string, valuesFileContent string) *C.char {
```

Comment:
Maybe we should think about how we want go code formatted

Reply:
There is a built-in formatter that can be called with `go fmt`, and we can set up its execution as part of build. However, it might make sense to do it with Gradle already

Reply:
Unfortunately spotless doesn't support Go code :-( I agree we should think about it.

Reply:
Spotless might be capable of checking license headers, but I didn't find a clear instruction on how to set it up

Reply:
We should create separate tickets that can be tackled after migrating to gradle

Reply:
Issue created: https://sonarsource.atlassian.net/browse/SONARIAC-1172 (out of scope of current sprint for now).